projects/kernel-config/README.md

```diff
@@ -0,0 +1,38 @@
+## kernel config project
+
+The intent of the kernel config project is to demonstrate a better way to
+handle kernel config. Specifically:
+
+* support for arch and version specific config
+* make diffs as readable as possible
+* ensure that all of our config settings are kept after oldconfig
+
+We achieve the goals by:
+
+* having version-specific config in separate files, which are automatically
+  merged
+* only keeping track of visible symbols, only keeping track of a delta from
```

Comment:
Sorry, "symbol" in this context means something like CONFIG_USB or whatever (this is what the kconfig code calls them). kconfig distinguishes between "visbile" and "invisible" symbols, "visible" symbols are ones that users can choose values for, and "invisible" symbols are ones that users can't (but have some value that's a result of a computation across other values).

Basically, since users don't control e.g. CONFIG_IRQ_WORK (it's invisible), there's no reason for us to track it in our diffs.